app/models/supplier_location.rb

```diff
@@ -6,6 +6,11 @@ class SupplierLocation < ApplicationRecord
 
   validate :effective_to_after_effective_from
 
+  scope :location, ->(location_id) { where(location_id: location_id) }
```

Comment:
Do we need to make the location a scope? that's so simple

Reply:
why not?

Reply:
Scope does make sense only with a meaningful name (like :cancelled, :active, published, ...)
The name "location" does not say anything about "the scope", the name is confusing, it sounds like a belongs_to relationship, 
if you want to keep it I'd suggest to rename it  with `with_location` or `having_location`, ... up to you :) 

`SupplierLocation.location(location.id)....`